manifests/record/a.pp

```diff
@@ -8,21 +8,32 @@
   $data,
   $ttl = '',
   $ptr = false,
-  $host = $name ) {
+  $host = $name,
+  $data_dir = $::dns::server::params::data_dir,
```

Comment:
All of these `params` references need to be `dns::server::config::foo`

params just is a class to provide defaults. Users are encouraged to set thing, but not via params, via the config class.